### PR TITLE
docs: improve contributing guide

### PR DESCRIPTION
**Pull request by jeromevdl**
**Issue #, if available:** #1295 

## Description of changes:

Improve the contributing guide with further content, especially checkstyle for code style.

**Checklist**

 

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda-java/#tenets)
* [ ] Update tests
* [x] Update docs
* [x] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,24 +1,134 @@
...
+[When filing an issue](https://github.com/aws-powertools/powertools-lambda-java/issues/new/choose), please check [existing open](https://github.com/aws-powertools/powertools-lambda-java/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc), or [recently closed](https://github.com/aws-powertools/powertools-lambda-java/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed) issues to make sure somebody else hasn't already reported the issue. Please try to include as much information as you can.
+<!-- markdownlint-enable MD013 -->
+
+## Contributing via Pull Requests
+
+Contributions via pull requests are much appreciated. Before sending us a pull request, please ensure that:
+
+1. You are working on a fork. [Fork the repository](https://github.com/aws-powertools/powertools-lambda-java/fork).
+2. You are working against the latest source on the **main** branch.
+3. You check existing open, and recently merged pull requests to make sure someone else hasn't addressed the problem already.
```
Comment (scottgerring) [inline CONTRIBUTING.md:42]:
```suggestion
3. You've check existing open, and recently merged pull requests to make sure someone else hasn't addressed the problem already.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,24 +1,134 @@
...
+4. You open an [issue](https://github.com/aws-powertools/powertools-lambda-java/issues/new/choose) before you begin any implementation. We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.
+
+### Dev setup
+
+We recommend using [IntelliJ IDEA](https://www.jetbrains.com/idea/) from Jetbrains.
```
Comment (kozub) [inline CONTRIBUTING.md:47]: The correct name of the company is `JetBrains` ;-)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,24 +1,134 @@
...
+        - [Dev setup](#dev-setup)
+        - [Local documentation](#local-documentation)
+    - [Conventions](#conventions)
+        - [General terminology and practices](#general-terminology-and-practices)
+        - [Testing definition](#testing-definition)
+    - [Finding contributions to work on](#finding-contributions-to-work-on)
+    - [Code of Conduct](#code-of-conduct)
+    - [Security issue notifications](#security-issue-notifications)
+    - [Troubleshooting](#troubleshooting)
+        - [API reference documentation](#api-reference-documentation)
```
Comment (kozub) [inline CONTRIBUTING.md:16]: Looks that we don't have `API reference doc` section so it can be removed from this content list.

**Review by kozub — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,24 +1,134 @@
...
+| ----------------- | ----------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------- |
+| Unit tests        | Verify the smallest possible unit works.                                                              | Networking access is prohibited. Prefer Functional tests given our complexity.                                                  | Lightning fast (nsec to ms)                       |
+| End-to-end tests  | Gain confidence that a Lambda function with our code operates as expected.                            | It simulates how customers configure, deploy, and run their Lambda function - Event Source configuration, IAM permissions, etc. | Slow (minutes)                                    |
+
+**NOTE**: Unit tests are mandatory. End-to-end tests should be created for new modules.
```
Comment (scottgerring) [inline CONTRIBUTING.md:108]: Something here about test pyramid? Otherwise we'll end up with a test cube, or something 😢

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,24 +1,134 @@
...
+## Troubleshooting
+
+
+## Licensing
+
+See the [LICENSE](LICENSE) file for our project's licensing. We will ask you to confirm the licensing of your contribution.
+
+We may ask you to sign a [Contributor License Agreement (CLA)](http://en.wikipedia.org/wiki/Contributor_License_Agreement) for larger changes.
 
 ## Contributing via Pull Requests
```
Comment (scottgerring) [inline CONTRIBUTING.md:133]: This section appears twice

Comment (scottgerring) [reply CONTRIBUTING.md:133]: In fact everything from here down is duplicated

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,59 +1,128 @@
...
+![](docs/media/intellij_checkstyle_1.png)
+
+3. Select the code you've created (module, package, class) and reformat code: `⌘⌥L` (macOS), or `Ctrl+Alt+L` (Windows/Linux):
+![](docs/media/intellij_checkstyle_2.png)
+
+4. Apply the reformat, optimize imports, rearrange and cleanup to your code and only to java files:
+![](docs/media/intellij_checkstyle_3.png)
+
+#### License headers
+All the java files should contain the licence/copyright header. You can copy past it from the [license-header](license-header) file.
```
Comment (eldimi) [inline CONTRIBUTING.md:65]:
```suggestion
All the java files should contain the licence/copyright header. You can copy paste it from the [license-header](license-header) file.
```

**Review by scottgerring — APPROVED** [review]
LGTM!

**Pull request merged**